zerver/lib/user_groups.py

```diff
@@ -569,7 +570,9 @@ def get_group_setting_value_for_register_api(
     return realm_setting_anonymous_group_membership[setting_group_id]
 
 
-def get_members_and_subgroups_of_groups(group_ids: set[int]) -> dict[int, UserGroupMembersDataDict]:
+def get_members_and_subgroups_of_groups(
+    group_ids: set[int],
+) -> dict[int, UserGroupMembersTypedDict]:
```

Comment:
This function was only being used at one place and it was not making use of the benefits that a dataclass provides. So replacing it with a typed dict seemed fine.